system/Router/RouteCollection.php

```diff
@@ -242,12 +248,22 @@ class RouteCollection implements RouteCollectionInterface
     /**
      * Constructor
      */
-    public function __construct(FileLocator $locator, Modules $moduleConfig)
+    public function __construct(FileLocator $locator, Modules $moduleConfig, Routing $routing)
```

Comment:
All of the tests are already passing related to this. 

In the same way the module config is sent it, I sent the Routing config in, so - seems like that's the proper way to do it....

Reply:
It would only break code for those that have extended the RouteCollection class, which I can't imagine is that many. If you think this particular BC is too much, we can scrap this and save it for 5. Seemed like a smaller BC possibility than a number of our other breaks, unless I'm completely missing something here.